src/dotnet/Chat.UI.Blazor/Components/ChatAuthorBadge/ChatAuthorBadgeBase.cs

```diff
@@ -32,33 +33,69 @@ public abstract class ChatAuthorBadgeBase : ComputedStateComponent<ChatAuthorBad
     }
 
     protected override ComputedState<Model>.Options GetStateOptions()
-        => new() { InitialValue = Model.None };
+    {
+        var initialValue = Model.None;
+        if (EvalInitialValue) {
+            var parsedChatAuthorId = new ParsedChatAuthorId(AuthorId).AssertValid();
+            var chatId = parsedChatAuthorId.ChatId.Id;
+            if (!chatId.IsEmpty) {
+                // Try to initialize with cached value
+                var authorTask = GetChatAuthor(Session, parsedChatAuthorId.ChatId, AuthorId, default);
+                if (authorTask.IsCompletedSuccessfully) {
+ #pragma warning disable VSTHRD002
+                    var author = authorTask.Result;
```

Comment:
why do we need to break the concept?

Reply:
I don't want to await the task. If it's completed task, then it means that I can get the value synchronously and use it for the initial render.